lighthouse.go

```diff
@@ -262,6 +262,19 @@ func (lh *LightHouse) reload(c *config.C, initial bool) error {
 
 	//NOTE: many things will get much simpler when we combine static_host_map and lighthouse.hosts in config
 	if initial || c.HasChanged("static_host_map") || c.HasChanged("static_map.cadence") || c.HasChanged("static_map.network") || c.HasChanged("static_map.lookup_timeout") {
+		// Clean up. Entries still in the static_host_map will be re-built.
+		// Entries no longer present must have their (possible) background DNS goroutines stopped.
+		if existingStaticList := lh.staticList.Load(); existingStaticList != nil {
+			lh.Lock()
+			for staticVpnIp := range *existingStaticList {
+				am := lh.unlockedGetRemoteList(staticVpnIp)
```

Comment:
I'm not sure how it would come up in practice but `unlockedGetRemoteList` has a side effect of adding an entry if it doesn't exist, this is why it needs a write lock.

We could check `lh.addrMap[staticVpnIp]` directly and use a read lock instead